avocado/plugins/runner_nrunner.py

```diff
@@ -233,29 +232,29 @@ def _get_all_runtime_tasks(test_suite):
         runtime_tasks = []
         test_result_total = test_suite.variants.get_number_of_tests(test_suite.tests)
         no_digits = len(str(test_result_total))
+        # define execution order
         execution_order = test_suite.config.get('run.execution_order')
         if execution_order == "variants-per-test":
-            for index, (runnable, variant) in enumerate(((test, variant)
-                                                         for test in test_suite.tests
-                                                         for variant in test_suite.variants.itertests()),
-                                                        start=1):
-                runtime_tasks.extend(Runner._create_runtime_tasks_for_test(
-                    test_suite,
-                    runnable,
-                    no_digits,
-                    index,
-                    variant))
+            test_variant = [(test, variant) for test in test_suite.tests
+                            for variant in test_suite.variants.itertests()]
         elif execution_order == "tests-per-variant":
-            for index, (runnable, variant) in enumerate(((test, variant)
-                                                         for variant in test_suite.variants.itertests()
-                                                         for test in test_suite.tests),
-                                                        start=1):
-                runtime_tasks.extend(Runner._create_runtime_tasks_for_test(
-                    test_suite,
-                    runnable,
-                    no_digits,
-                    index,
-                    variant))
+            test_variant = [(test, variant)
+                            for variant in test_suite.variants.itertests()
+                            for test in test_suite.tests]
+        # decide if a copy of the runnable is needed, in case of more variants
+        # than tests
+        copy_runnable = False
+        if len(test_variant) > len(test_suite.tests):
+            copy_runnable = True
```

Comment:
There is also the possibility of creating a copy of the runnable here, avoiding the new parameter on `_create_runtime_tasks_for_test`. Let me know what you prefer.

Reply:
Let me know your opinion here, @clebergnu, which may change the behavior and affect your previous comment.

Reply:
> Let me know your opinion here, @clebergnu, which may change the behavior and affect your previous comment.

Sounds better to me indeed.

Reply:
As it was a small change, I did a force-push.